geppetto.js/geppetto-ui/src/3d-canvas/showcase/examples/SelectionUtils.js

```diff
@@ -0,0 +1,40 @@
+import { hexToRGBNormalized } from "../../threeDEngine/util";
+
+const SELECTION_COLOR = { r: 0.8, g: 0.8, b: 0, a: 1 };
+
+
+export function dataMapping (data){
+  return data.map(item => (
+    {
+      color: item.selected ? SELECTION_COLOR : item.color ? item.color : hexToRGBNormalized(GEPPETTO.Resources.COLORS.DEFAULT),
+      instancePath: item.instancePath
+    }
+  ))
+}
+
+export function onSelection (selectedInstances) {
+  const { data } = this.state
+  const newData = []
+  let newInstance = true
+  for (const si of selectedInstances){
+    for (const i of data){
+      if (si === i.instancePath){
+        newData.push({
+          ...i,
+          selected: !i.selected
+        })
+        newInstance = false
+      } else {
+        newData.push({ ...i, })
+      }
+    }
+    if (newInstance){
```

Comment:
Why do we use the `newInstance` flag here? 🤔 

I see that it adds a new instance if the selected instance is not part of the data.
But how can we have instances here that are not part of the data?

Reply:
This relates to the vfb showcase example.

Let's call 'parent' instance to an instance that inherits the visual capability from its child, like:
![image](https://user-images.githubusercontent.com/19196034/125063722-bd98cc80-e0a7-11eb-91a7-cb7a1349ddff.png)

In the canvas proxy Instances, we can specify these 'parent' instances, and what the mesh factory algorithm does is traverse the 'parent' instance tree and display everything the 'parent' has to show (**including what his children have to show**).
We end up with meshes that are coming from the 'child' instances being displayed. 

Upon selection:
(data are the proxy instances, selectedInstances are the meshes I clicked)
![image](https://user-images.githubusercontent.com/19196034/125063954-06508580-e0a8-11eb-99af-8fbb98facf48.png)

There are pertinent reasons to use the current behavior instead of just seeing what we specify, like not having to specify 40 individual instances that belong to the same population (which is what happens on acnet examples)

Reply:
okay I see, then let's keep it for now. The way I understood is that the Canvas should display what it gets, so we should refactor this in the future, not removing it but rather wrapping it around the Canvas.